src/riak_kv_bucket.erl

```diff
@@ -43,6 +44,43 @@
 
 -export_type([props/0]).
 
+-define(VALID_PROPERTIES,
+        ["allow_mult",
+         "basic_quorum",
+         "big_vclock",
+         "bucket_type",
+         "chash_keyfun",
+         "dvv_enabled",
+         "dw",
+         "last_write_wins",
+         "linkfun",
+         "n_val",
+         "notfound_ok",
+         "old_vclock",
+         "postcommit",
+         "pr",
+         "precommit",
+         "pw",
+         "r",
+         "rw",
+         "small_vclock",
+         "w",
+         "write_once",
+         "young_vclock",
+         %% TS-specific ones:
+         "ddl",
+         "table_def"
+        ]).
+
+-spec is_valid_property(string() | binary()) -> boolean().
```

Comment:
Tough call. We've not constrained bucket properties in the past, and it's unclear this is a good reason to start doing so.

On the other hand, we've certainly wanted to document all possible bucket properties to get out of the JSON-on-the-CLI business for riak-admin bucket-type create, so this work may prove valuable even if we choose not to use it.